### PR TITLE
Yan MB-6163 Dynamic Weight Allowance On the Move 101

### PR DESCRIPTION
**Pull request by YanZ777**
## Description

Added the DBAuthorizedWeight to the payload so that it could be read into MovingInfo. 
Prevented the Alert from displaying if there is no authorized Weight.

## Reviewer Notes

Since there is old data that doesn't have the DBAuthorizedWeight on the entitlements, the decision was made to not show the alert on the Move 101 page.
https://ustcdp3.slack.com/archives/CTNTFJSBA/p1610482483495400

There's existing header violations inside of aXe that I am not addressing in this PR.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make client_run
make server_run
```
Create a new service member and go through the workflow until you hit the Move 101 page.
Also refresh the Move 101 page. 
Login as sm_no_move_type@example.com and check the Move 101 page.
Make sure the weight allowance matches the rank inside of entitlements.go

## Code Review Verification Steps

(https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [x] There are no aXe warnings for UI.
* [x] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [x] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6163) for this change

### CODE REVIEW

File: src/pages/MyMove/MovingInfo.jsx [hunk 1]
```diff
@@ -1,100 +1,152 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
```
Comment (suzubara) [inline src/pages/MyMove/MovingInfo.jsx:2]: We don't need to use `bindActionCreators`, you can assign `mapDispatchToProps` to be an object instead of a function ie:
```
const mapDispatchToProps = {
  fetchLatestOrders: fetchLatestOrdersAction
}
```

File: src/pages/MyMove/MovingInfo.jsx [hunk 2]
```diff
@@ -1,100 +1,152 @@
...
+};
+
+MovingInfo.defaultProps = {
+  entitlementWeight: 0,
 };
 
-export default MovingInfo;
+function mapStateToProps(state, ownProps) {
+  const { moveId } = ownProps.match.params;
+  const orders = selectOrdersForMove(state, moveId);
```
Comment (suzubara) [inline src/pages/MyMove/MovingInfo.jsx:132]: Do you need to use the `moveId` to select the orders? I think you should be able to use `selectCurrentOrders` to get the orders for the logged in service member.. then you wouldn't need the `moveId` and don't have to add `withRouter`

File: swagger/internal.yaml [hunk 3]
```diff
@@ -2190,6 +2190,10 @@ definitions:
         x-nullable: true
       department_indicator:
         $ref: '#/definitions/DeptIndicator'
+      dBAuthorizedWeight:
```
Comment (suzubara) [inline swagger/internal.yaml:2193]: the naming of this field (prefix with `dB`) is strange to me, is there a reason to not call it `authorizedWeight`?

File: pkg/testdatagen/make_entitlement.go [hunk 4]
```diff
@@ -26,6 +26,8 @@ func MakeEntitlement(db *pop.Connection, assertions Assertions) models.Entitleme
 		StorageInTransit:      &storageInTransit,
 	}
 	entitlement.SetWeightAllotment(*grade)
+	dBAuthorizedWeight := entitlement.AuthorizedWeight()
+	entitlement.DBAuthorizedWeight = dBAuthorizedWeight
```
Comment (YanZ777) [inline pkg/testdatagen/make_entitlement.go:30]: @donaldthai @kimallen 
I noticed that these lines were causing some CircleCI tests to fail. I've tested both with and without these lines of code, and the Move 101 page still gets the weights dynamically. If that is the case, are these lines ok to take out?

File: swagger/internal.yaml (hunk 3 above)
Comment (YanZ777) [reply swagger/internal.yaml:2193]: I may be remembering incorrectly, but I think I was told that the internal.yaml has to match up with something else.
Maybe @kimallen can weigh in on this more?

File: pkg/handlers/internalapi/orders.go [hunk 5]
```diff
@@ -29,6 +29,14 @@ func payloadForOrdersModel(storer storage.FileStorer, order models.Order) (*inte
 		moves = append(moves, payload)
 	}
 
+	var dBAuthorizedWeight *int64
+	dBAuthorizedWeight = nil
+	if order.Entitlement != nil {
+		if order.Entitlement.DBAuthorizedWeight != nil {
+			dBAuthorizedWeight = swag.Int64(int64(*order.Entitlement.DBAuthorizedWeight))
+		}
+	}
```
Comment (donaldthai) [inline pkg/handlers/internalapi/orders.go:38]: Hey, the authorized weight can be simplified by calling the `AuthorizedWeight()` function: https://github.com/transcom/mymove/blob/master/pkg/models/ghc_entitlements.go#L41

File: swagger/internal.yaml (hunk 3 above)
Comment (donaldthai) [reply swagger/internal.yaml:2193]: There is a property named `DBAuthorizedWeight` that refers to the `authorized_weight` column in the `ghc_entitlements.go` file but I do think it would make more sense to rename to `authorizedWeight` since `DBAuthorizedWeight` is meant to be used to determine if the column value was set in the DB.

Comment (donaldthai) [reply swagger/internal.yaml:2193]: `AuthorizedWeight()` in `ghc_entitlements.go` determines the value as well

Comment (kimallen) [reply swagger/internal.yaml:2193]: What I remember is that when we tried to use just `authorizedWeight` in the yaml, then it didn't match the existing field on Entitlement called `DBAuthorizedWeight`, so it was erroring out.  Maybe there's another way to rename than what we were trying.

File: src/pages/MyMove/MovingInfo.test.jsx [hunk 6]
```diff
@@ -3,12 +3,20 @@ import { mount } from 'enzyme';
...
+const wrapper = mount(<MovingInfo {...testProps} />);
 
 describe('MovingInfo component', () => {
   it('renders', () => {
     expect(wrapper.find('MovingInfo').length).toBe(1);
     expect(wrapper.text()).toContain('Tips for planning your shipments');
+    expect(wrapper.text()).toContain('7,000 lbs');
     expect(wrapper.find('[data-testid="shipmentsHeader"]').length).toBe(1);
     expect(wrapper.find('[data-testid="shipmentsSubHeader"]').length).toBe(4);
   });
```
Comment (suzubara) [inline src/pages/MyMove/MovingInfo.test.jsx:22]: It would be good to add a test case if `entitlementWeight` is `0` or even null/undefined to make sure this page doesn't crash

**Review by suzubara — APPROVED** [review]
left a suggestion for adding another test case, but otherwise looks good!

**Review by suzubara — APPROVED** [review]
F/E changes look good to me
